File: client/src/app/models/content.ts
import { Albumn, AlbumnRequest } from "./albumn";
import { RegisterSong } from "./song";

export class Content {
  id: string | undefined;
  hostName: string | undefined;
  date: Date | undefined;
  address: string | undefined;
  story: string | undefined;
  wish: string | undefined;
  personName: string | undefined;
}

export class ContentRequest {
  id: string | undefined;
  hostName: string | undefined;
  date: Date | undefined;
  address: string | undefined;
  story: string | undefined;
  wish: string | undefined;
  personName: string | undefined;
  album: AlbumnRequest[] | undefined;
  registerSong: RegisterSong[] | undefined;
}

File: server/WebApplication1/WebApplication1/Models/Content.cs
using System.ComponentModel.DataAnnotations;
namespace WebApplication1.Models
{
    public class Content
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string HostName { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public string Story { get; set; }

        [Required]
        public string Wish { get; set; }

        [Required]
        public int PersonId { get; set; }

        public Person Person { get; set; }

        public List<Albumn> Albumns { get; set; }

        public List<RegisterSong> RegisterSongs { get; set; }
    }
}
